app/services/query.rb

```diff
@@ -0,0 +1,12 @@
+class Query
```

Comment:
Could possibly be `BaseQuery` or something I guess. Or even a module that's mixed in where needed.

Reply:
Sounds good, module for now.